design/token-mechanics-v2.1.md

```diff
@@ -1,46 +1,55 @@
-
-  3.3- How we can make sure there is enough tokens to pay successful challenges?
-  > We should decide whether we want to accept defaults or not, i.e. do we want to consider an scenario where a package does not have enough 
-    tokens staked to pay a challenge? If not, we should pay a proportional amount of tokens based on the severity and the total stake.
-    
-  3.4- How we will handle a huge list of vouchers during a payout?
-  > We can have a rate per version in ZEP that will be adjust each time a challenge is solved.  
+# Token Mechanics v2.1
+
+## Protocol
+
+### Vouching
+
+- Developers will register plain addresses. These addresses may represent anything: an EVM package, an EVM package version, a logic contract from a package, a contract instance, or a mock address representing something off-chain. 
+- Registering a new entry requires a minimum stake that cannot be removed, and is used for spam prevention. The minimum stake can only be moved to another entry owned by the same developer, but never withdrawn from the vouching system.
+- Any user can vouch for an entry, and withdraw or move their vouching without penalizations.
+
+### Challenging
+
+- Any user can challenge an entry, staking an amount of tokens behind the challenge. The amount of tokens associated to the challenge should be proportional to the vouching of the entry and the severity of the challenge, and these values should be tabulated off-chain.
+- A challenge can be accepted or rejected by the owner of the challenged entry. Regardless of the result, it enters a period of N days during which it can be appealed. If not appealed, it is then confirmed and executed. Appeals allow challengers to have their rejected challengers reviewed, and prevent dishonest developers from colluding with a challenger to drain funds from the other vouchers of an entry.
+- Any user can appeal a resolution of a challenge (during the appealing period) by staking tokens behind it. Resolution of appeals will be centralized initially, and then solved via a PLCR voting by all ZEP token holders.
+
+## Issues
+
+1- How we can avoid users unvouching their tokens given a new challenge yet unsolved?
+  
+  We can lock all funds that are needed to pay for all open challenges, and allow users to unvouch the rest. Note that this may not be implementable due to gas constraints. Another simple option is to outright lock all funds on a challenged entry. 
+  
+  We may also allow users to signal their willingness to retreat from a package, in which case new challenges do not apply to them, and once the last challenge to their funds ends, then they are allowed to retreat.ges
+
+2- How can we prevent a user from grabbing an off-chain challenge and submitting it under their name before the original auditor?
+
+  We can handle front-running scenarios by manually checking both challenges and deciding which is the real one based on off-chain information.
```

Comment:
IMO this scenario is really hard to solve, we may not have access to all the information necessary to solve such a case.